channel/script.go

```diff
@@ -509,9 +651,145 @@ func (ctrl *npcChatController) SendGuildEmblemEditor() {
 // SendShop packet to Player
 func (ctrl *npcChatController) SendShop(goods [][]int32) {
 	ctrl.goods = goods
+	ctrl.persistShop = true
```

Comment:
I am surprised you need to do this. I would expect when the client closes the shop it would send a finished npc chat packet.